spring-kafka/src/main/java/org/springframework/kafka/support/micrometer/MicrometerHolder.java

```diff
@@ -69,12 +75,56 @@ public MicrometerHolder(@Nullable ApplicationContext context, String name,
 		this.timerDesc = timerDesc;
 		this.name = name;
 		this.tags = tags;
+		registries = filterRegistries(registries, context);
 		if (registries.size() == 1) {
 			this.registry = registries.values().iterator().next();
 			buildTimer(NONE_EXCEPTION_METERS_KEY);
 		}
 		else {
-			throw new IllegalStateException("No micrometer registry present (or more than one)");
+			throw new IllegalStateException("No micrometer registry present (or more than one and "
+					+ "none marked @Primary)");
+		}
+	}
+
+	private Map<String, MeterRegistry> filterRegistries(Map<String, MeterRegistry> registries,
+			ApplicationContext context) {
+
+		if (registries.size() == 1) {
+			return registries;
+		}
+		Map<String, MeterRegistry> filtered = registries.entrySet()
+				.stream()
+				.filter(entry -> entry.getValue()
+						.getClass()
+						.getName()
+						.equals("org.springframework.boot.actuate.autoconfigure.metrics."
+								+ "AutoConfiguredCompositeMeterRegistry"))
```

Comment:
Why do we need this if it is `@Primary` in the auto-configuration:
```
	@Bean
	@Primary
	AutoConfiguredCompositeMeterRegistry compositeMeterRegistry(Clock clock, List<MeterRegistry> registries) {
		return new AutoConfiguredCompositeMeterRegistry(clock, registries);
	}
```
?

I mean it looks like the next logic covers this Spring Boot "pollution" as well.

Do I miss anything ?

Thanks

Reply:
Oh; thanks; I didn't look at that; I just saw 3 beans in my test app; will test and push.